src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -486,18 +487,61 @@ void checkConstraintLanguageVersionAndRuntimeSupportForConversion(SyntaxNode syn
                 }
                 else if (conversion.IsSpan)
                 {
+                    Debug.Assert(destination.OriginalDefinition.IsSpan() || destination.OriginalDefinition.IsReadOnlySpan());
+
+                    CheckFeatureAvailability(syntax, MessageID.IDS_FeatureFirstClassSpan, diagnostics);
+
                     // PROTOTYPE: Check runtime APIs used for other span conversions once they are implemented.
-                    if (destination.OriginalDefinition.Equals(Compilation.GetWellKnownType(WellKnownType.System_ReadOnlySpan_T), TypeCompareKind.AllIgnoreOptions))
+                    // NOTE: We cannot use well-known members because per the spec
+                    // the Span types involved in the Span conversions can be any that match the type name.
+                    if (TryFindImplicitOperatorFromArray(destination.OriginalDefinition) is { } method)
                     {
-                        _ = GetWellKnownTypeMember(WellKnownMember.System_ReadOnlySpan_T__op_Implicit_Array, diagnostics, syntax: syntax);
+                        diagnostics.ReportUseSite(method, syntax);
                     }
                     else
                     {
-                        Debug.Assert(destination.OriginalDefinition.Equals(Compilation.GetWellKnownType(WellKnownType.System_Span_T), TypeCompareKind.AllIgnoreOptions));
-                        _ = GetWellKnownTypeMember(WellKnownMember.System_Span_T__op_Implicit_Array, diagnostics, syntax: syntax);
+                        Error(diagnostics,
+                            ErrorCode.ERR_MissingPredefinedMember,
+                            syntax,
+                            destination.OriginalDefinition,
+                            WellKnownMemberNames.ImplicitConversionName);
+                    }
+                }
+            }
+        }
+
+        internal static MethodSymbol? TryFindImplicitOperatorFromArray(TypeSymbol type)
+        {
+            return TryFindSingleMember(type, WellKnownMemberNames.ImplicitConversionName,
+                static (member) => member is MethodSymbol
+                {
+                    ParameterCount: 1,
+                    Arity: 0,
+                    IsStatic: true,
+                    DeclaredAccessibility: Accessibility.Public,
+                    Parameters: [{ Type: ArrayTypeSymbol { IsSZArray: true, ElementType: TypeParameterSymbol } }]
+                } method ? method : null);
+        }
+
+        private static T? TryFindSingleMember<T>(TypeSymbol type, string name, Func<Symbol, T?> predicate) where T : class
```

Comment:
><T> [](http://example.com/codeflow?start=45&length=3)

Is the type parameter necessary, or could this be simplified to `TryFindSingleMethod` with `MethodSymbol` directly?